src/hooks/useGeneratedKey.ts

```diff
@@ -0,0 +1,37 @@
+//Keep track of all the keys we've generated to ensure uniqueness
+let usedKeys: Array<string> = [];
```

Comment:
I think we would need to keep a reference to this array, not entirely sure 🧐. I would probably move it inside the function useGeneratedKey with a 

```tsx
const usedKeys = useRef<Array<string>>([]);
```

and then further used as `usedKeys.current.push()`